llama_index/llms/base.py

```diff
@@ -337,3 +348,117 @@ async def astream_complete(
         self, prompt: str, **kwargs: Any
     ) -> CompletionResponseAsyncGen:
         """Async streaming completion endpoint for LLM."""
+
+    def _as_query_component(self, **kwargs: Any) -> QueryComponent:
+        """Return query component."""
+        if self.metadata.is_chat_model:
+            return LLMChatComponent(llm=self)
+        else:
+            return LLMCompleteComponent(llm=self)
+
+
+class LLMCompleteComponent(QueryComponent):
+    """LLM completion component."""
+
+    llm: BaseLLM = Field(..., description="LLM")
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def set_callback_manager(self, callback_manager: Any) -> None:
+        """Set callback manager."""
+        self.llm.callback_manager = callback_manager
+
+    def _validate_component_inputs(self, input: Dict[str, Any]) -> Dict[str, Any]:
+        """Validate component inputs during run_component."""
+        if "prompt" not in input:
+            raise ValueError("Prompt must be in input dict.")
+        # do special check to see if prompt is a list of chat messages
+        if isinstance(input["prompt"], get_args(List[ChatMessage])):
+            input["prompt"] = messages_to_prompt(input["prompt"])
```

Comment:
So the input to an LLM can never be a list of chat messages? That seems not ideal for APIs like openai?

(also, best to use `self.llm.messages_to_prompt()`)